extension/src/functions.rs

```diff
@@ -462,7 +476,7 @@ mod tests {
 				Ok(Converging(0))
 			));
 			// Check if the contract environment buffer is written correctly.
-			assert_eq!(env.buffer, UppercaseConverter::convert(expected, &env));
+			assert_eq!(env.buffer, UppercaseConverter::try_convert(expected, &env).unwrap());
```

Comment:
```suggestion
			assert_eq!(Ok(env.buffer), UppercaseConverter::try_convert(expected, &env));
```
Just a minor suggestion, if we `unwrap()`, the test will panics because the error returned from `try_convert`. But if we compare the `Result`, test will throw `assert_eq!` error with comparison between two values which, imo, is a better experience.